packages/edit-site/src/components/global-styles/font-library-modal/context.js

```diff
@@ -192,35 +203,96 @@ function FontLibraryProvider( { children } ) {
+			const alreadyInstalledFontFaces =
+				installedFontFamily.fontFace.filter( ( fontFaceToInstall ) =>
+					checkFontFaceInstalled(
+						fontFaceToInstall,
+						fontFamilyToInstall.fontFace
+					)
+				);
+
+			// Filter out Font Faces that have already been installed (so that they are not re-installed)
+			fontFamilyToInstall.fontFace = fontFamilyToInstall.fontFace.filter(
+				( fontFaceToInstall ) =>
+					! checkFontFaceInstalled(
+						fontFaceToInstall,
+						installedFontFamily.fontFace
+					)
+			);
+
 			// Install the fonts (upload the font files to the server and create the post in the database).
-			const response = await fetchInstallFont( formData );
-			const fontsInstalled = response?.successes || [];
-			// Get intersecting font faces between the fonts we tried to installed and the fonts that were installed
-			// (to avoid activating a non installed font).
-			const fontToBeActivated = getIntersectingFontFaces(
-				fontsInstalled,
-				[ font ]
+			let sucessfullyInstalledFontFaces = [];
+			let unsucessfullyInstalledFontFaces = [];
+			if ( fontFamilyToInstall.fontFace.length > 0 ) {
+				const response = await batchInstallFontFaces(
+					installedFontFamily.id,
+					makeFontFacesFormData( fontFamilyToInstall )
+				);
+				sucessfullyInstalledFontFaces = response?.successes;
+				unsucessfullyInstalledFontFaces = response?.errors;
+			}
+
+			const detailedErrorMessage = unsucessfullyInstalledFontFaces.reduce(
+				( errorMessageCollection, error ) => {
+					return `${ errorMessageCollection } ${ error.message }`;
+				},
+				''
 			);
-			// Activate the font families (add the font families to the global styles).
-			activateCustomFontFamilies( fontToBeActivated );
+
+			// If there were no successes and nothing already installed then we don't need to activate anything and can bounce now.
+			if (
+				sucessfullyInstalledFontFaces.length === 0 &&
+				alreadyInstalledFontFaces.length === 0
+			) {
+				throw new Error(
+					__( 'No font faces were installed. ' ) +
+						detailedErrorMessage
+				);
+			}
+
+			// Use the sucessfully installed font faces
+			// As well as any font faces that were already installed (those will be activated)
+			fontFamilyToInstall.fontFace = [
+				...sucessfullyInstalledFontFaces,
+				...alreadyInstalledFontFaces,
+			];
+
+			// Activate the font family (add the font family to the global styles).
+			activateCustomFontFamilies( [ fontFamilyToInstall ] );
+
 			// Save the global styles to the database.
 			saveSpecifiedEntityEdits( 'root', 'globalStyles', globalStylesId, [
 				'settings.typography.fontFamilies',
 			] );
+
 			refreshLibrary();
-			setIsInstalling( false );
 
-			return response;
-		} catch ( error ) {
+			if ( unsucessfullyInstalledFontFaces.length > 0 ) {
+				throw new Error(
+					__(
+						'Some font faces were installed. There were some errors. '
+					) + detailedErrorMessage
+				);
```

Comment:
This should use sprintf and use translation placeholders.

Reply:
Addressed in a separate PR here: https://github.com/WordPress/gutenberg/pull/57961/commits/37f31520f905a6450e756c2f8003474f92a51880